[docs] Update guidance about Colima/Docker Desktop requirements and Mutagen (#4694)
drud/ddev: docs/content/users/install/ddev-installation.md

    You can include a `-s <version>` argument to install a specific release or a prerelease version:

    ```
    curl -fsSL https://raw.githubusercontent.com/drud/ddev/master/scripts/install_ddev.sh | bash -s v1.21.5
    ```

>     Most macOS users will want to enable Mutagen for superb performance; no installation is required, just `ddev config global --mutagen-enabled`.

Reviewer: I’d... 

- Drop the redundant “macOS” since we’re in a macOS-specific section. 
- Make a strong recommendation so it’s not about popularity and we’re not leaving someone to wonder what macOS users _wouldn’t_ want to enable Mutagen. 
- Drop the explanation of steps that _don’t_ need to be taken. (Installation not required.)
- Replace “superb” with “the best” because it’s more about the best chance of great performance without being any kind of qualitative guarantee.
- Link to the broader explanation of the feature for context.
- Link the command to start reiterating how commands (and the documentation structure) work
- Remove that “just” and “run” the command [for consistency](https://ddev.readthedocs.io/en/stable/developers/writing-style-guide/).

```suggestion
    We recommend [enabling Mutagen](performance.md#mutagen) for the best performance, by running [`ddev config global --mutagen-enabled`](../usage/commands.md#config).
```
